src/PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.cpp

```diff
@@ -0,0 +1,92 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/ForceFree/ExactWald.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/ForceFree/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ForceFree::Solutions {
+
+ExactWald::ExactWald(const double magnetic_field_amplitude,
+                     const Options::Context& /*context*/)
+    : magnetic_field_amplitude_(magnetic_field_amplitude) {}
+
+ExactWald::ExactWald(CkMigrateMessage* msg) : InitialData(msg) {}
+
+std::unique_ptr<evolution::initial_data::InitialData> ExactWald::get_clone()
+    const {
+  return std::make_unique<ExactWald>(*this);
+}
+
+void ExactWald::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | background_spacetime_;
+  p | magnetic_field_amplitude_;
+}
+
+PUP::able::PUP_ID ExactWald::my_PUP_ID = 0;
```

Comment:
Just NOLINT this to make clang-tidy happy.  We can't do anything about charm internals.